packages/flutter/test/widgets/image_test.dart

```diff
@@ -1981,3 +2024,44 @@ class FailingImageProvider extends ImageProvider<int> {
     );
   }
 }
+
+class FailSuccessFailImageProvider extends ImageProvider<Object> {
+  FailSuccessFailImageProvider({this.image, this.throws});
+
+  final Object throws;
+  final ui.Image image;
+
+  @override
+  Future<Object> obtainKey(ImageConfiguration configuration) {
+    return SynchronousFuture<FailSuccessFailImageProvider>(this);
+  }
+
+  @override
+  void resolveStreamForKey(ImageConfiguration configuration, ImageStream stream,
+      Object key, ImageErrorListener handleError) {
+    super.resolveStreamForKey(configuration, stream, key, handleError);
+  }
+
+  int _loadCallCount = 0;
+
+  @override
+  ImageStreamCompleter load(Object key, DecoderCallback decode) {
+    _loadCallCount += 1;
+    // Fail on first load, succeed on second and fails again on third load.
+    if (_loadCallCount == 1 || _loadCallCount == 3) {
+      throw throws;
```

Comment:
@dnfield I can't understand why the test case throw this on second call.

Reply:
I'm not clear on what you mean.

Reply:
Ah I see the test now. Perhaps you could instrument this method to print the load count locally and figure out who's calling it?

Reply:
Thanks for this. I found out that the Image widget was calling it.